### PR TITLE
Issue 99 Allow user to add SOL tokens via account token list

### LINKED ISSUE #99: Allow user to add SOL tokens via account token list
A user should be able to go to the account token list and click an add button that will prompt the user to enter a mint address and allow for the user to create an associatedTokenAccount for that address. Then we will need to show the token metadata and balance in the token list.

### PR DESCRIPTION
**Pull request by dannpl**
Closes #99

https://www.loom.com/share/4ec16124267449d5af4d5fb81bca7ff5

### CODE REVIEW

File: src/storage/TokensProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+const initialState = {
+  tokens: [],
+  handleUpdateTokens: () => {},
+  visibleTokens: [],
+  isActiveToken: () => false,
+}
+
+const TokensContext =
+  createContext<ReturnType<typeof useTokensHook>>(initialState)
```
Comment (Perronef5) [inline src/storage/TokensProvider.tsx:252]: We are moving the app towards using `redux`. Do you think we could benefit off of making this a slice or adding this as part of an existing slice?

File: src/utils/solanaUtils.ts [hunk 2]
```diff
@@ -139,6 +139,48 @@ export const readSolanaBalance = async (cluster: Cluster, address: string) => {
...
+  try {
+    const response = await getOrCreateAssociatedTokenAccount(
+      connection,
+      signer,
+      mintPublicKey,
+      payerPublicKey,
+    )
+
+    return response
+  } catch {}
```
Comment (Perronef5) [inline src/utils/solanaUtils.ts:181]: We could log some error here to make it easier to debug

File: .github/workflows/solana-build-workflow.yml [hunk 3]
```diff
@@ -14,12 +14,12 @@ jobs:
           # The branch to sync from
```
[the remark this answers is not in the document]
Comment (dannpl) [reply .github/workflows/solana-build-workflow.yml]: Done!

File: src/features/account/AccountTokenList.tsx [hunk 4]
```diff
@@ -1,142 +1,76 @@
...
+  const { visibleTokens } = useTokens()
 
   const bottomSpace = useMemo(() => bottom * 2, [bottom])
 
-  const tokens = useMemo(() => {
-    if (loading) {
-      return []
-    }
+  const filteredTokens = useMemo(() => {
+    if (updatingTokens || loading) return []
```
[the remark this answers is not in the document]
Comment (dannpl) [reply src/features/account/AccountTokenList.tsx]: Done, i add the skeleton only in loading.

File: src/features/account/manageToken/AccountManageTokenAddScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+          {INPUTS.map((item) => (
+            <TextInput
+              key={item.value}
+              backgroundColor="secondaryBackground"
+              marginBottom="m"
+              height={56}
+              paddingHorizontal="m"
+              borderRadius="m"
+              fontSize={18}
```
[the remark this answers is not in the document]
Comment (dannpl) [reply src/features/account/manageToken/AccountManageTokenAddScreen.tsx]: Done!

File: src/features/account/manageToken/AccountManageTokenAddScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+  )
+}
+
+export default AccountManageTokenAddScreen
+
+//
+// Utils
+//
+
+const INPUTS: {
```
[the remark this answers is not in the document]
Comment (dannpl) [reply src/features/account/manageToken/AccountManageTokenAddScreen.tsx]: Done!

File: src/features/account/manageToken/AccountManageTokenAddScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+              />
+            </Box>
+          </Box>
+        </Box>
+      </Box>
+    </Box>
+  )
+}
+
+export default AccountManageTokenAddScreen
```
[the remark this answers is not in the document]
Comment (dannpl) [reply src/features/account/manageToken/AccountManageTokenAddScreen.tsx]: Done!

File: src/features/account/manageToken/AccountManageTokenListScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      <FlatList
+        data={tokens}
+        renderItem={renderItem}
+        keyExtractor={keyExtractor}
+      />
+    </Box>
+  )
+}
+
+export default AccountManageTokenListScreen
```
[the remark this answers is not in the document]
Comment (dannpl) [reply src/features/account/manageToken/AccountManageTokenListScreen.tsx]: Done!

File: src/storage/TokensProvider.tsx [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+      ),
+    [currentAccount, restoredVisibleTokens],
+  )
+
+  useAsync(async () => {
+    try {
+      const response = await restoreVisibleTokens()
+
+      setRestoredVisibleTokens(response)
+    } catch {}
```
[the remark this answers is not in the document]
Comment (dannpl) [reply src/storage/TokensProvider.tsx]: Done!

File: src/storage/TokensProvider.tsx [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+              ),
+              name: item.name,
+              staked: false,
+            })
+          }
+        })
+
+        return newTokens
+      })
+    } catch {}
```
[the remark this answers is not in the document]
Comment (dannpl) [reply src/storage/TokensProvider.tsx]: Added, thankss!

File: src/utils/solanaUtils.ts (hunk 2 above)
Comment (dannpl) [reply src/utils/solanaUtils.ts]: Done!

**Comment by ChewingGlass** [comment]
Closing this PR, a new PR with a large refactor accomplishes this

**Pull request closed without merging**